en/editor/publish/custom-build-plugin.md

```diff
@@ -44,92 +54,124 @@ export const configs: IConfigs = {
+
+1. The environment variables in different processes will be different. The entry script will be loaded by the rendering process and the main process at the same time, do not use the editor interface that only exists in a single process in the entry script.
 
-详细的接口定义说明如下：
+2. There are two ways to configure the key of `config`: one is for a single platform configuration, and the key is filled in as **platform plugin name** (available in the editor menu bar **Extensions -> Extension Manager -> Built-in** To view the platform plug-in name); one is the configuration for all platforms, the key is filled in as `*`. These two configuration methods are mutually exclusive, please do not use them in the same build extension package.
+
+Example:
 
 ```ts
 declare type IConfigs = Record<Platform | '*', IPlatformConfig>;
 declare interface IBuildPlugin {
-    hooks?: string; // 钩子函数的存储路径
-    options?: IDisplayOptions; // 需要注入的平台参数配置
-    verifyRuleMap?: IVerificationRuleMap; // 注册参数校验规则函数
+    hooks?: string; // Storage path of hook function
+    options?: IDisplayOptions; // Platform parameter configuration that needs to be injected
+    verifyRuleMap?: IVerificationRuleMap; // Register parameter verification rule function
 }
 declare type IDisplayOptions = Record<string, IConfigItem>;
 declare interface IConfigItem {
-    // 默认值，注册的默认值将会在插件自身配置里的 options.[platform].xxx 字段内
+    // The default value, the registered default value will be in the options.[platform].xxx field in the plugin configuration
     default?: any;
 
     render: ?{
-        // 渲染 ui 组件规则，与 ui-prop 处统一规则一致，只有指定了 ui 属性的配置才会在构建配置面板上显示
+        // The rules for rendering ui components are consistent with the unified rules at ui-prop. Only configurations with ui properties specified will be displayed on the build configuration panel
         ui?: string;
-        // 传给 ui 组件的配置参数
+        // The configuration parameters passed to the ui component
         attributes?: IUiOptions;
     };
 
-    // 配置显示的名字，如果需要翻译，则传入 i18n:${key}
+    // Configure the displayed name, if you need to translate, then pass in i18n:${key}
     label?: string;
 
-    // 设置的简单说明，将会显示在配置名称的鼠标上移 title 上
+    // A brief description of the setting will be displayed on the title of the configuration name
     description?: string;
 
-    // 配置的类型
+    // Type of configuration
     type?: 'array' | 'object';
 
-    // 如果 type 是 array，则会按照指定数据类型和 itemConfigs 来渲染数据
+    // If type is an array, the data will be rendered according to the specified data type and itemConfigs
     itemConfigs?: Record<string, IConfigItem> | IConfigItem[];
 }
 
 declare interface IUiOptions extends IOptionsBase {
-    // 校验规则数组，构建提供一些基础规则，也可以通过 verifyRuleMap 来指定新的校验规则，只有当传入 require 时才会做无值的校验，否则仅存在值时才校验
+    // Validation rule array, build to provide some basic rules, you can also specify a new validation rule through verifyRuleMap, only when pass in require will be a valueless check, otherwise only when there is a value
```

Comment:
```suggestion
    // Validation rules array, build provides some basic rules, and you can also specify new validation rules through “verifyRuleMap”. Only when pass in “require” will be a valueless checksum, otherwise only when there is a value.
```